backend/utils/discounts.js

```diff
@@ -182,6 +182,8 @@ const validateDiscountOnOrder = async (
   }, 0)
 
   const shipping = get(cart, 'shipping.amount', 0)
+  const taxRate = parseFloat(get(cart, 'taxRate', 0))
+  const totalTaxes = Math.ceil(taxRate * subTotal)
```

Comment:
Sadly I don't think we have any unit test for calculating discounts. Would be great to add some :)

Another suggestion would be to keep cleaning the utils directory by moving this to logic/discount - really imho utils should only be for low level utilities.

Reply:
Will add them